internal/ncproxy/store/store.go

```diff
@@ -0,0 +1,279 @@
+			return err
+		}
+		jsonEndptData, err := json.Marshal(endpt)
+		if err != nil {
+			return err
+		}
+		return bkt.Put([]byte(endpt.EndpointName), jsonEndptData)
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (n *NetworkingStore) DeleteEndpoint(ctx context.Context, endpointName string) error {
+	if err := n.db.Update(func(tx *bolt.Tx) error {
+		bkt := getEndpointBucket(tx)
+		if bkt == nil {
+			return errors.Wrapf(ErrBucketNotFound, "bucket %v", bucketKeyEndpoint)
+		}
+		return bkt.Delete([]byte(endpointName))
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (n *NetworkingStore) ListEndpoints(ctx context.Context) (results []*ncproxynetworking.Endpoint, err error) {
+	if err := n.db.View(func(tx *bolt.Tx) error {
+		bkt := getEndpointBucket(tx)
+		if bkt == nil {
+			return errors.Wrapf(ErrBucketNotFound, "endpoint bucket %v", bucketKeyEndpoint)
+		}
+		err := bkt.ForEach(func(k, v []byte) error {
+			jsonData := bkt.Get([]byte(k))
+			if jsonData == nil {
+				return errors.Wrapf(errKeyNotFound, "endpoint %v", k)
+			}
+			endptInternal := &ncproxynetworking.Endpoint{}
+			if err := json.Unmarshal(jsonData, endptInternal); err != nil {
+				return err
+			}
+			results = append(results, endptInternal)
+			return nil
+		})
+		return err
+	}); err != nil {
+		return nil, err
+	}
+
+	return results, nil
+}
+
+// ComputeAgentStore is a database that stores a key value pair of container id
+// to compute agent server address
+type ComputeAgentStore struct {
+	DB *bolt.DB
+}
+
+func NewComputeAgentStore(db *bolt.DB) *ComputeAgentStore {
+	return &ComputeAgentStore{DB: db}
+}
+
+func (c *ComputeAgentStore) Close() error {
+	return c.DB.Close()
+}
+
+// GetComputeAgent returns the compute agent address of a single entry in the database for key `containerID`
+// or returns an error if the key does not exist
+func (c *ComputeAgentStore) GetComputeAgent(ctx context.Context, containerID string) (result string, err error) {
+	if err := c.DB.View(func(tx *bolt.Tx) error {
+		bkt := getComputeAgentBucket(tx)
+		if bkt == nil {
+			return errors.Wrapf(ErrBucketNotFound, "bucket %v", bucketKeyComputeAgent)
+		}
+		data := bkt.Get([]byte(containerID))
+		if data == nil {
+			return errors.Wrapf(errKeyNotFound, "key %v", containerID)
+		}
+		result = string(data)
+		return nil
+	}); err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+// GetComputeAgents returns a map of the key value pairs stored in the database
+// where the keys are the containerIDs and the values are the corresponding compute agent
+// server addresses
+func (c *ComputeAgentStore) GetComputeAgents(ctx context.Context) (map[string]string, error) {
+	content := map[string]string{}
+	if err := c.DB.View(func(tx *bolt.Tx) error {
+		bkt := getComputeAgentBucket(tx)
+		if bkt == nil {
+			return errors.Wrapf(ErrBucketNotFound, "bucket %v", bucketKeyComputeAgent)
+		}
+		err := bkt.ForEach(func(k, v []byte) error {
+			data := bkt.Get([]byte(k))
```

Comment:
Isn't `v` from the function parameter the data, so there isnt a need re-`Get` it from the bucket?